### DISCUSSION TITLE
클린 아키텍처와 모듈분리

### DISCUSSION THREAD

**Discussion by KimGyeongyeon** [boostcampwm-2022/android08-KiwiTalk#12] [프로젝트 회고]
클린아키텍처 적용에 대해 논의했다
<img width="300" alt="구조A" src="https://user-images.githubusercontent.com/67995865/202193047-27b68d98-36aa-41a8-9cdf-45a8123a376c.png">
<img width="450" alt="구조B" src="https://user-images.githubusercontent.com/67995865/202193053-a139c0ae-c6e1-4042-88dc-aca25f6b7cec.png">
<img width="450" alt="구조C" src="https://user-images.githubusercontent.com/67995865/202193055-2c8be6d7-ae5c-4fb0-b632-a9295923554c.png">

**Comment by KimGyeongyeon** [comment]
의존성 역전에 대해 더 알아보면 좋을 것 같다.
interface를 생성하는 이유를 한 단어로 어렵게 말하는 것으로 예상됨
https://meetup.toast.com/posts/345

**Comment by SteelZoo** [comment]
시간을 많이 썼지만 이유를 알고, 근거를 갖고 결과를 냈기 때문에 좋은 대화였다
[[그룹프로젝트]우리가 domain 모듈을 도입한 과정](https://steelzoodev.tistory.com/8)

**Comment by KimGyeongyeon** [comment]
app 모듈에서 다른 모듈(data)에게 Hilt로 종속성 주입하기

- 우선 project, app gradle에 힐트를 적용했다.
- 근데 data 모듈에 hilt 종속성 코드가 없는 조가 있었다.
- buildSrc를 사용해서 중복되는 종속성 관리를 통합한 것임을 깨달았다.
- Kts migration + buildSrc 사용 대신 주석을 잘 다는 것으로 유지보수를 대체하기로 했다.  익숙한 문법을 사용할 수 있고, 보일러 플레이트 코드를 줄일 수 있다는 장점이 있었지만, 팀원 모두가 모르는 상태라 당장 적용하기에는 장점에 비해 리스크가 컸다.

결론 : 그대로 Groovy 사용. Inject가 필요한 모듈의 gradle에 `implementation "javax.inject:javax.inject:1"` 적용. (모듈 생성 계획은 아직 없으니까)

**Comment by KimGyeongyeon** [comment]
## 기술 선택 모범 답안 흐름

### 멀티모듈이란?


### 공부해본 멀티모듈의 장단점

### 대체 가능한 방식

### 논의 과정
app와 data로 처음 모듈을 나눴는데, data 모듈에 domain과 data를 통합해서 넣었을 때 domain을 Kotlin module로 안드로이드 의존성이 전혀 없이 뺄 수 있다는 것을 깨달았다. 
나눌수록 멀티모듈의 장점인 빌드 속도도 더욱 줄어들고 역할분리도 명확해지기 때문에 domain을 모듈로 분리하되, Usecase는 과도한 추상화라고 생각해서 제외했다.

### 적용시에 느낀 실질적 장점
